cms/static/js/factories/import.js

```diff
@@ -33,39 +51,36 @@ define([
             maxChunkSize: 20 * 1000000, // 20 MB
             autoUpload: false,
             add: function(e, data) {
-                Import.clearImportDisplay();
-                Import.okayToNavigateAway = false;
+                Import.reset();
                 submitBtn.unbind('click');
+
                 file = data.files[0];
+
                 if (file.name.match(/tar\.gz$/)) {
-                    submitBtn.click(function(event){
+                    submitBtn.click(function(event) {
                         event.preventDefault();
-                        $.cookie('lastfileupload', file.name);
+
+                        Import.start(
+                            file.name,
+                            feedbackUrl.replace('fillerName', file.name)
+                        ).then(onComplete);
+
                         submitBtn.hide();
-                        Import.startUploadFeedback();
-                        data.submit().complete(function(result, textStatus, xhr) {
-                            window.onbeforeunload = null;
-                            if (xhr.status != 200) {
+                        data.submit().complete(function (result, textStatus, xhr) {
+                            if (xhr.status !== 200) {
                                 var serverMsg, errMsg, stage;
+
                                 try{
-                                    serverMsg = $.parseJSON(result.responseText);
+                                    serverMsg = $.parseJSON(result.responseText) || {};
                                 } catch (e) {
                                     return;
                                 }
-                                errMsg = serverMsg.hasOwnProperty('ErrMsg') ?  serverMsg.ErrMsg : '' ;
-                                if (serverMsg.hasOwnProperty('Stage')) {
-                                    stage = Math.abs(serverMsg.Stage);
-                                    Import.stageError(stage, defaults[stage] + errMsg);
-                                }
-                                else {
-                                    alert(gettext('Your import has failed.') + '\n\n' + errMsg);
-                                }
-                                chooseBtn.html(gettext('Choose new file')).show();
-                                bar.hide();
+
+                                errMsg = serverMsg.hasOwnProperty('ErrMsg') ? serverMsg.ErrMsg : '';
+                                stage = Math.abs(serverMsg.Stage || 0);
```

Comment:
Ok so the error handling part here is quite tricky so bear with me a second:

In the `master` branch I found out that there are two flaws in how `factories/import.js` handles the errors: a logic flaw and a code flaw.

The logic flaw is the following:
on `data.submit().complete()`, if `xhr.status` is `200` then the code checks if the server response contains a `Stage` property, otherwise it displays an alert. It's all good until you realize that, after you start an import and you pass the first two stages, if you refresh the page then you are basically aborting the request to the server, which in turns respond with a `xhr.status` equal to `0` and without the `Stage` parameter. As a consequence you get the "import has failed" alert even if you are not supposed to!

You notice though that you don't actually get the alert, and the reason is because of the code flaw:
```javascript
// if you are losing the connection or refreshing the page
// parseJSON returns an empty string
serverMsg = $.parseJSON(result.responseText);
// ...

// error as the empty string doesn't have that method!
serverMsg.hasOwnProperty('ErrMsg'); 
```
so basically you have a JS error that is preventing the wrong alert dialog to pop up!

This is what happens now on `master`:
![before-code-error](https://cloud.githubusercontent.com/assets/6400898/7833962/4b456860-046b-11e5-8e0f-479a2439624a.gif)

This is what would happen if the code error is fixed, leaving everything else untouched:
![before-no-code-error](https://cloud.githubusercontent.com/assets/6400898/7833967/616425aa-046b-11e5-9121-a37a80d0fd1d.gif)

---

So I tried again to improve how we deal with the errors since my first attempt wasn't as good as it should've been. 

The main point is to be able to understand if, when we get an error response with no data from the server, it's because the server actually can't tell what's going on or if it's simply because the user has refreshed the page. 

What I did then was to flag when the request was about to be aborted because of the user refresh via `onbeforeunload` and now it seems to work as intended (even f it's difficult what would that be since it was broken to begin with :D)

Here is a before/after example of what happens if you lose connection before submitting the file (so the scenario that you created):

Before
![before-when-no-connection](https://cloud.githubusercontent.com/assets/6400898/7834047/fe39b0e8-046b-11e5-87e4-7b443d7d9900.gif)

After
![after-when-no-connection](https://cloud.githubusercontent.com/assets/6400898/7834048/03f3c532-046c-11e5-8372-5437f4b0f8b1.gif)

---

I hope I managed to explain the issue clearly, and of course please let me know what you think :)